rfcs/328-declarative-configuration.md

```diff
@@ -0,0 +1,149 @@
+# Declarative Configuration
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-goals](#non-goals)
+- [Design Details](#design-details)
+  - [Repository Configuration](#repository-configuration)
+  - [Magefile Targets Implementation Details](#magefile-targets-implementation-details)
+  - [Magefile Targets Available Day One](#magefile-targets-available-day-one)
+  - [Testing Framework](#testing-framework)
+  - [Configuration Examples](#configuration-examples)
+- [Testing and Release Plan](#testing-and-release-plan)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+
+## Summary
+
+This proposal replaces environment variable based configuration being used in Make+Magefiles in favor of using declarative yaml based configuration.
+
+## Motivation
+
+As we've been writing `devbase`, and other tooling that uses it, we've identified that the current practice of environment variable based configuration as well as opinionated and unchangeable defaults is not flexible enough for using this tooling both outside of stencil-base (where it is primarily used today) and giving our users the flexibility to easily make minor behaviour changes. This has, ultimately created two pain points:
+
+ * Documentation is hard to write and not easily discoverable
+ * Finding configuration consumption, and configurable areas is difficult
+ * Using `devbase` outside of `stencil-base` projects is not easy (must use sub-modules, and even then it's hard to "plug and play") 
+
+### Goals
+
+ - All configuration should be written as go structs (as we intend to only write new targets in), and have godoc compatible documentation on each exported field along with examples as needed.
+ - Standard location for all configuration for tooling provided by devbase, et. al
```

Comment:
Gr, I'm mad about that. George had me remove that since it was leaking context about internal usage @ Outreach. I agree with you though, I'll workshop this